### PR TITLE
Added CLTV/CSV functionality

### PR DESCRIPTION
**Pull request by jimmysong**
This makes the CLTV/CSV stuff functional and actually evaluates everything properly. Lots of new tests, also had to change how witness stuff was being serialized and interpreted.

### CODE REVIEW

**Comment by jimmysong** [comment]
CLTV/CSV is now available as part of Taproot trees by specifying a locktime or sequence.

**Comment by jimmysong** [comment]
Added gracefully degrading multisig (k-of-n then after some time, k-1 of n and then k-2 of n until 1 of n)

File: buidl/op.py [hunk 1]
```diff
@@ -784,97 +845,47 @@ def op_checkmultisigverify(stack, tx_obj, input_index):
     return op_checkmultisig(stack, tx_obj, input_index) and op_verify(stack)
 
 
-def op_checklocktimeverify(stack, locktime, sequence):
-    if sequence == 0xFFFFFFFF:
+def op_checklocktimeverify(stack, tx_obj, input_index):
+    sequence = tx_obj.tx_ins[input_index].sequence
+    locktime = tx_obj.locktime
```
Comment (mflaxman) [inline buidl/op.py:850]: Should this method be part of the Tx class?

**Review by mflaxman — APPROVED** [review]
Didn't have time to dig into the details yet, but in the interim wanted to say that this looks REALLY cool!

File: buidl/op.py (hunk 1 above)
Comment (jimmysong) [reply buidl/op.py:850]: CLTV is an op, though.

**Comment by jimmysong** [comment]
@mflaxman can I merge this?

**Pull request merged**